docs/chain/did.md

```diff
@@ -0,0 +1,73 @@
+# DID 
+
+## What is a DID?
+
+In KILT a DID is a decentralized identifier that the user owns and controls. It consists of a unique set of keys that can be used for different operations on the blockchain. For an in-depth explanation see the [KILT DID spec](https://github.com/KILTprotocol/kilt-did-driver/blob/master/docs/did-spec/spec.md).
+
+A DID may be a "light" DID, which is not stored on-chain, or a "full" on-chain DID. A light DID is issued by default, with the keys stored locally on your device. By upgrading this to a full DID registered on the blockchain, all the keys associated with it can be retrieved from the KILT blockchain storage.
+
+A full DID can then be used to perform certain on-chain actions which include:
+
+* Writing CTypes to the chain
+* Writing attestations to the chain
+* Setting delegations
+* Doing key rotations on the DID keys
+
+## Registering a DID
+
+A full DID is needed if the user wants to become an attester or wants to setup delegations. A full DID also allows the user to embed a list of URLs, known as service endpoints, into the DID document so that it can be retrieved from the chain.
+
+To create a full DID the user first has to create a set of keys and service endpoints:
+
+* one authentication key for signing extrinsics from your DID
+* zero or more key agreement keys for encrypting messages that are sent to you
+* one attestation key for signing attestations (optional)
+* one delegation key for authorizing delegations (optional)
+* service endpoints that point to external hostings for others to find (optional)
+
+With these they are ready to write the DID to the KILT blockchain.
+
+The user then has to create the `did::create` extrinsic and sign it with any KILT account that has enough funding to pay both the transaction fees and the  DID deposit. The extrinsic consists of  
```

Comment:
I would maybe start with explaining the DID Creation Details.

In this section you started with explaining the content of the DID creation details. So I would keep the top down approach.

1. Keys and Service Endpoints form the Creation details 
2. Creation details and signature build the create extrinsic
3. create extrinsic needs to be signed by someone who pays the fees and deposit.